### DISCUSSION TITLE
About packaging bash files into PyPI

### DISCUSSION THREAD

**Discussion by Zengyf-CVer** [pypa/setuptools#3123] [Q&A]
I have a bash file (*.sh) in my project and there is a python file that needs to call this .sh file. When I packaged it with setuptools, I found that the .sh file was not packaged, only the .py file was packaged. How should I pack the .sh file into it?

**Comment by abravalheri** [comment]
Hi @Zengyf-CVer, have you tried the ["data-files"](https://setuptools.pypa.io/en/latest/userguide/datafiles.html) approach?

Personally, I would place the bash files inside the package directory, use `setuptools-scm` to make sure all files in the git repository are added to the sdist, set `include_package_data=True`, and finally use `importlib.resources` (or `importlib_resources)` to load the file in runtime.
